_posts/en/newsletters/2019-05-14-newsletter.md

```diff
@@ -0,0 +1,485 @@
+version of their SHA256 function that's been pre-initialized with the
+prefix tag so it doesn't need to repeat those operations for every
+signature verification.  Implementations that don't require maximal
+performance (such as ordinary wallets) can just implement the algorithm
+as described using their default SHA256 library function.
+
+There are also some changes to what data is included in the hash and how
+it is serialized.  This includes improvements that can help make wallets
+without access to verified UTXO entries (e.g. hardware wallets and
+[HSMs][]) more efficient because they don't need to retrieve as much
+data in order to ensure they're signing for correct UTXOs and amounts.
+
+Although that may sound like a significant number of changes, we suspect
+it's only an afternoon worth of work making minor serialization changes
+for any wallet that's already segwit compatible and which can gain
+access to a library like [libsecp256k1][] for generating
+bip-schnorr-compatible signatures.
+
+## Simple multisig spending using Taproot
+
+After single-sig UTXOs, the most common are simple multisig UTXOs.
+These are outputs that depend on a certain number of signatures from
+particular pubkeys but don't have any other conditions.  These are used
+both by individual users (e.g. requiring multiple devices to work
+together in a spend) and by multiple parties to a single transaction
+(e.g. escrows and LN).
+
+There are two ways to perform simple multisig spending using Taproot,
+the most efficient of which is key and signature aggregation as
+described in this subsection.  We'll examine the other mechanism in the
+next subsection.
+
+For aggregation, two or more private keys are created and their pubkeys
+are derived.  The pubkeys are then combined into a single pubkey that's
+indistinguishable from any other Bitcoin pubkey.  This is used as the
+segwit witness program (e.g. bech32 address) as described in the
+previous subsection.  Later, the owner or owners of some or all of the
+private keys create signatures that are combined into a single signature
+that's also indistinguishable from any other bip-schnorr signature.
+This must commit to the same transaction digest as described in the
+previous subsection, but the result is a multisig spend that's
+completely indistinguishable from a single-sig spend.
+
+You may have noticed that the preceding paragraph is vague about the
+exact mechanism used to aggregate the keys and signatures.  The reason
+for that is because there are multiple known methods and any one of them
+can be used by the participants.  It's even possible for researchers to
+find new methods and implement them in Bitcoin wallets without any
+consensus changes.  This is because the signature algorithm is only
+looking for a single pubkey and a single signature that are valid under
+rules described in the single-sig subsection above.  That said, of the
+methods known, the [MuSig protocol][] is probably the best studied
```

Comment:
Right, I was hoping to avoid those details here because they're not directly relevant to the bip-taproot proposal.  The point I want to make is that you can use any protocol you want as long is it produces a valid sig for a valid pubkey.  There are ways known for creating aggregated pubkeys and signatures without interaction---they're just not secure against your counter signers.  Maybe that isn't a problem for some application, so they can use that protocol.  Or, as you say, maybe improvements will be made.  I don't think those specifics matter in the context of bip-taproot; what matters is that we know aggregation is possible and that it's up to individiual application developers to choose how they want to do it and what tradeoffs they want to make.